### ISSUE TITLE
Running cost sensitive classification - data input and output command line

### ISSUE THREAD

**Issue by dhruvghulati-zz** [andreasvlachos/arow_csc#3]
I am attempting to use `arow.py` for multi class cost sensitive classification.

My training data (`training.dat`) looks like:

`1:0.5 2:0.4 3:0.12 4:0.78 6:0.90 7:0.2 8:0.57 9:0.5 10:0.3 11:0.9 12:0.8 13:0.14 14:0.43 15:0.1 16:0.54 0| location_slot number_slot`

Where `1:0.5` says that I have cost 0.5 of assigning label 1 to training instance 0. The features are in string format, and not sparse (just bag of words). 

And test data (`test.dat`) looks like:

`0| population location_slot estimated united nations number_slot placed number population among nations world`

Where this is instance 0 of my test set and the things after the `|` are my non-sparse, non vectorized features.
- How do I need to change my training and test data formats to conform with `arow.py`? 
- Do my costs need to all be between `[0..1]`? 
- How can I access the array of test prediction labels `arow.py` produces (is this  `[cl.predict(d, verbose=True).label for d in data]` in your test file?)
- The labels seem to be either `pos` or `neg` in your test case - how do I adapt for multi-class?

**Comment by andreasvlachos** [comment]
I think the best way to achieve what you need is to write a bit of code to load your data (the one in the main function is just meant to be an example). You need to modify this bit more or less to load the training data: https://github.com/andreasvlachos/arow_csc/blob/master/arow.py#L529-L549 Essentially, all you need is to read in the class costs and then the feature weights. To load the test data it should be the same but without the costs per class.

Answering the other questions:
- costs can be 0 or any positive real.
- to predict use this function: https://github.com/andreasvlachos/arow_csc/blob/master/arow.py#L529-L549 and then access the prediction object returned.
- The example is binary, but the labels are just strings. Can create as many classes as needed, the training cost vector per instance is just a python dictionary (https://github.com/andreasvlachos/arow_csc/blob/master/arow.py#L532).

**Comment by dhruvghulati-zz** [comment]
Understood. So this line:

`costs["neg"]` could be adapted to `costs[3]=0.12` etc for all of my 16 labels in the instance above, and I can ignore the if condition (if -1 or +1)?

For the features within `featureID, featureVal = feature.split(":")`, and `featureVector[featureID] = float(featureVal)`, it seems like you have a dictionary where the keys are just the index of the feature, and the `featureVal` is converted to a float.

Does this mean I cannot use strings (bag of words) as features, but I could use sparse vector transforms of the features i.e. `[0 1 0 0 0 0 0 ]`?

**Comment by dhruvghulati-zz** [comment]
Another quick question, because you say `testingInstances = instances[int(len(instances) * 0.75) + 1:]`, this means that the test instances are taken from the same array as the training instances are taken from (i.e. you also had costs for the test instances). Do I need to have costs for my test instances, or can I just have features as I do currently?

**Comment by dhruvghulati-zz** [comment]
My questions still stand (will add to better README) but I am live now, everything is working.

**Comment by andreasvlachos** [comment]
You can use bag of words features, that's exactly what the dictionary structure used is good for and you can use the predict function to predict instances that were not in the training data and thus no cost vector is needed for them. If you want to write a tutorial on how to use the classifier I would be happy to help. Perhaps a jupyter notebook would be good.
